### PR TITLE
Feat/add driver fe

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- Defined constants for managing driver data.
- Added functionality to create new drivers, including a dedicated API endpoint.
- Integrated i18n to handle multilingual text for driver-related UI.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/browse/SCRUM-121)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [ ] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

photo_2024-12-03_15-53-43
photo_2024-12-03_15-53-50

### CODE REVIEW

File: src/constants/validation.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const phoneRegex = /^\+380\d{9}$/;
+
```
Comment (Parle648) [inline src/constants/validation.ts:2]: consts upperCase

File: src/pages/CreateRoute/DriversStage/index.tsx [hunk 2]
```diff
@@ -12,16 +12,30 @@ import SortDriversRow from './components/SordDriversRow';
 import { useExportDrivers } from './useGetDrivers';
 import { useSearchDrivers } from './useSearchDrivers';
 import { useChooseDriver } from './useChoseDriver';
+import DriverForm from './components/DriverForm';
+import { useTranslation } from 'react-i18next';
```
Comment (Parle648) [inline src/pages/CreateRoute/DriversStage/index.tsx]: 1.1.14

File: src/pages/CreateRoute/DriversStage/styles.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { SxProps } from '@mui/system';
+import { Theme } from '@mui/material/styles';
+
+export const driversFormContainer: SxProps<Theme> = {
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  p: 3,
```
Comment (Parle648) [inline src/pages/CreateRoute/DriversStage/styles.ts]: ?

File: src/pages/CreateRoute/DriversStage/index.tsx [hunk 4]
```diff
@@ -12,16 +12,30 @@ import SortDriversRow from './components/SordDriversRow';
...
+import { addDrivers } from './api/getDrivers';
+import { driversFormContainer } from './styles';
 
 const DriversStagePage: FC = () => {
-  const { drivers } = useExportDrivers();
+  const { t } = useTranslation();
```
Comment (Parle648) [inline src/pages/CreateRoute/DriversStage/index.tsx]: just import { t } from 'i18....

Comment (oyakovytskyi) [reply src/pages/CreateRoute/DriversStage/index.tsx]: okay done

File: src/constants/validation.ts (hunk 1 above)
Comment (oyakovytskyi) [reply src/constants/validation.ts:2]: done

File: src/pages/CreateRoute/DriversStage/index.tsx (hunk 2 above)
Comment (oyakovytskyi) [reply src/pages/CreateRoute/DriversStage/index.tsx]: done

File: src/pages/CreateRoute/DriversStage/styles.ts (hunk 3 above)
Comment (oyakovytskyi) [reply src/pages/CreateRoute/DriversStage/styles.ts]: make changes

**Review by Parle648 — APPROVED** [review]

**Review by lyudad — APPROVED** [review]

**Pull request merged**